agent/engine/docker_image_manager.go

```diff
@@ -379,70 +384,75 @@ func (imageManager *dockerImageManager) getNonECSContainerIDs(ctx context.Contex
 	return nonECSContainersIDs, nil
 }
 
-type ImageWithSize struct {
+type ImageWithSizeID struct {
 	ImageName string
+	ImageID   string
 	Size      int64
 }
 
 func (imageManager *dockerImageManager) removeNonECSImages(ctx context.Context, nonECSImagesNumToDelete int) {
 	if nonECSImagesNumToDelete == 0 {
 		return
 	}
-	var nonECSImageNames = imageManager.getNonECSImageNames(ctx)
-	var nonECSImageNamesRemoveEligible []string
-	for _, nonECSImage := range nonECSImageNames {
-		if !isInExclusionList(nonECSImage, imageManager.imageCleanupExclusionList) {
-			nonECSImageNamesRemoveEligible = append(nonECSImageNamesRemoveEligible, nonECSImage)
+	nonECSImages := imageManager.getNonECSImages(ctx)
+	var nonECSImagesRmEligible []ImageWithSizeID
```

Comment:
nit: don't think we need the word 'remove' or 'rm' in it since its inside `removeNonECSImages` method.

Reply:
👍